lib/rules/selector-max-compound-selectors/__tests__/index.mjs

```diff
@@ -384,3 +384,116 @@ testRule({
 		},
 	],
 });
+
+// Testing ignoreSelectors
+testRule({
+	ruleName,
+	config: [2, { ignoreSelectors: ['::v-deep', '/-moz-.*/', /-screen$/, '.ignored'] }],
+
+	accept: [
+		{
+			code: '::v-deep .foo .bar {}',
+			description: 'ignored selector at the beginning',
+		},
+		{
+			code: '.foo ::v-deep .bar {}',
+			description: 'ignored selector in the middle',
+		},
+		{
+			code: '.foo ::v-deep > .bar {}',
+			description: 'ignored selector in the middle followed by a combinator',
+		},
+		{
+			code: '.foo::v-deep .bar {}',
+			description: 'ignored selector as pseudo-element',
+		},
+		{
+			code: '.foo::v-deep > .bar {}',
+			description: 'ignored selector as pseudo-element followed by a combinator',
+		},
+		{
+			code: '.foo > ::v-deep .bar {}',
+			description: 'ignored selector preceded by a combinator',
+		},
+		{
+			code: '.foo input::-moz-placeholder {}',
+			description: 'selector ignored by string regex',
+		},
+		{
+			code: '.foo :-webkit-full-screen a {}',
+			description: 'selector ignored by regex',
+		},
+		{
+			code: '.foo::v-deep :-webkit-full-screen a ::-moz-placeholder {}',
```

Comment:
If you do the refactor, this would become `.foo ::v-deep input ::-moz-placeholder {}`.

Reply:
I have the same reason as https://github.com/stylelint/stylelint/pull/7544#discussion_r1524024793